vendor: add google impersonate package

rhu713 opened the pull request:
Adds the google impersonate package in anticipation of supporting assume role
functionality in GCP.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/81780)
<!-- Reviewable:end -->

Commit:
vendor: add google impersonate package

Adds the google impersonate package in anticipation of supporting assume role
functionality in GCP.

Release note: None

rhu713 commented:
bors r+
